src/Alpha_complex/test/Delaunay_complex_unit_test.cpp

```diff
@@ -0,0 +1,72 @@
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2020 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE "delaunay_complex"
+#include <boost/test/unit_test.hpp>
+#include <boost/mpl/list.hpp>
+
+#include <CGAL/Epick_d.h>
+#include <CGAL/Epeck_d.h>
+
+#include <vector>
+#include <limits>  // NaN
+#include <cmath>
+
+#include <gudhi/Alpha_complex.h>
+// to construct a simplex_tree from Delaunay_triangulation
+#include <gudhi/graph_simplicial_complex.h>
+#include <gudhi/Simplex_tree.h>
+#include <gudhi/Unitary_tests_utils.h>
+#include <gudhi/random_point_generators.h>
+
+// Use dynamic_dimension_tag for the user to be able to set dimension
+typedef CGAL::Epeck_d< CGAL::Dynamic_dimension_tag > Exact_kernel_d;
+// Use static dimension_tag for the user not to be able to set dimension
+typedef CGAL::Epeck_d< CGAL::Dimension_tag<5> > Exact_kernel_s;
+// Use dynamic_dimension_tag for the user to be able to set dimension
+typedef CGAL::Epick_d< CGAL::Dynamic_dimension_tag > Inexact_kernel_d;
+// Use static dimension_tag for the user not to be able to set dimension
+typedef CGAL::Epick_d< CGAL::Dimension_tag<5> > Inexact_kernel_s;
+// The triangulation uses the default instantiation of the TriangulationDataStructure template parameter
+
+typedef boost::mpl::list<Exact_kernel_d, Exact_kernel_s, Inexact_kernel_d, Inexact_kernel_s> list_of_kernel_variants;
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(Alpha_complex_from_OFF_file, TestedKernel, list_of_kernel_variants) {
+  std::cout << "*****************************************************************************************************";
+  using Point = typename TestedKernel::Point_d;
+  std::vector<Point> points;
+  // 50 points on a 4-sphere
+  points = Gudhi::generate_points_on_sphere_d<TestedKernel>(10, 5, 1.);
+
+  Gudhi::alpha_complex::Alpha_complex<TestedKernel> alpha_complex(points);
+
+  // Alpha complex
+  Gudhi::Simplex_tree<> stree_from_alpha_complex;
+  BOOST_CHECK(alpha_complex.create_complex(stree_from_alpha_complex));
+  stree_from_alpha_complex.initialize_filtration();
+
+  // Delaunay complex
+  Gudhi::Simplex_tree<> stree_from_delaunay_complex;
+  BOOST_CHECK(alpha_complex.create_complex(stree_from_delaunay_complex, 0., false, true));
+
+  // Check all the simplices from alpha complex are in the Delaunay complex
+  for (auto f_simplex : stree_from_alpha_complex.filtration_simplex_range()) {
+    std::vector<Gudhi::Simplex_tree<>::Vertex_handle> simplex;
+    for (Gudhi::Simplex_tree<>::Vertex_handle vertex : stree_from_alpha_complex.simplex_vertex_range(f_simplex)) {
+      std::cout << "(" << vertex << ")";
+      simplex.push_back(vertex);
+    }
+    std::cout << std::endl;
+    Gudhi::Simplex_tree<>::Simplex_handle sh = stree_from_delaunay_complex.find(simplex);
```

Comment:
Can't we pass `simplex_vertex_range(f_simplex)` directly to `find`? It is a range.

Reply:
I don't think so because here we have 2 different simplex trees. The first one is constructed as an alpha complex and the second as a delaunay complex. This means the simplex handles (iterator over the dictionnary) won't be the same for the 2 data structures. I just check I have the same triangulations in the 2 simplex trees, and filtration values are `NaN` for the Delaunay complex

Reply:
I didn't say to use the Simple_handle of one complex in the other. All I am saying is that simplex_vertex_range returns a range (of Vertex_handle, i.e. something like int), and that you are copying it to a vector, just to pass it to another function as a range. The intermediate vector seems unnecessary. Am I missing something?

Reply:
Ok, I got it. I removed the vector copy. It was just here in case it fails, but it shall not.